src/stream/src/executor/join/builder.rs

```diff
@@ -158,6 +158,7 @@ impl<const T: JoinTypePrimitive, const SIDE: SideTypePrimitive> JoinChunkBuilder
     }
 
     pub fn post_process(c: StreamChunk) -> StreamChunk {
+        c.check_consistency();
```

Comment:
We check before doing post_process... but it does not panic here, meaning there's no trailing `UpdateDelete`.

Reply:
It panics later on, when converting back from `StreamChunkMut` to `StreamChunk`: https://github.com/risingwavelabs/risingwave/pull/20341/files#diff-477cd761851ca4cd7ad5a34567afc8b2b4271f9c9096a0aa854f68847d5ad369R200

However I can't find anything in the logic below that can cause a hanging `UpdateDelete`.

Reply:
Oh nevermind. There can be a hanging update delete because the visibility is marked to be blank...